test/simulator/timestamp/call_log_manager/call_log_manager.cpp

```diff
@@ -146,6 +146,37 @@ call_log_manager::process_call_log_entry(json call_log_entry)
             }
             break;
         }
+        case api_method::begin_transaction: {
+            const std::string session_id = call_log_entry["session_id"].get<std::string>();
+            /*
+             * Not having a valid connection is a fatal error since no other operations can happen
+             * without a connection.
+             */
+            if (_conn == nullptr)
+                throw std::runtime_error("Could not close the session (session ID: " + session_id +
+                  ")" + ", connection does not exist");
+
+            /* We should not begin transactions on sessions with an ID that does not exist in the
+             * session map. */
+            if (_session_map.find(session_id) == _session_map.end()) {
+                std::cerr << "Could not begin transaction, session does not exist (session ID: "
+                          << session_id << ")" << std::endl;
+                break;
```

Comment:
Do we want to simply break here or exit?

Reply:
I guess we could just exit here to save other errors being printed if we continue to try doing other operations on an invalid session. I can change close session to reflect this too.

Reply:
By exit do you mean throw an error?